src/openrct2/ride/coaster/HybridCoaster.cpp

```diff
@@ -490,9 +562,29 @@ namespace HybridRC
         paint_session* session, uint16_t rideIndex, uint8_t trackSequence, uint8_t direction, int32_t height,
         const TileElement* tileElement)
     {
-        sub_98197C_rotated(
-            session, direction, GetTrackColour(session) | (SPR_G2_HYBRID_TRACK_VERTICAL + direction + 4), 0, 0, 32, 20, 3,
-            height, 0, 6, height + 8);
+        switch (direction)
+        {
+            case 0:
+                sub_98197C_rotated(
```

Comment:
The point of sub_xxyyzz_rotated() is not to have to write out all the directions. If you do need a separate call for each direction, then you should use the non-rotated version - but make sure to adjust the parameters accordingly if you do.

Reply:
The bounding boxes depend on the direction. That was what was causing the bug, I replaced the switch statement with a single call and that resulted in it using the same bounding boxes for every angle.

Reply:
I know, but using the `_rotated` calls while also splitting out per direction is just unnecessary overhead. You can achieve what you want using regular `sub_98197C()` calls.

But let's wait until Duncan has his solution ready, as it might address this issue in another way.